src/MSBuild.Abstractions/MSBuildWorkspace.cs

```diff
@@ -157,7 +158,42 @@ private BaselineProject CreateSdkBaselineProject(string projectFilePath, IProjec
                 propertiesInTheBaseline = propertiesInTheBaseline.Add(DesktopFacts.UseWPFPropertyName);
             }
 
-            return new BaselineProject(newProject, propertiesInTheBaseline, projectStyle);
+            return new BaselineProject(newProject, propertiesInTheBaseline, projectStyle, outputType);
+        }
+
+        private bool IsSupportedOutputType(ProjectOutputType type) =>
+            type switch
+            {
+                ProjectOutputType.Exe => true,
+                ProjectOutputType.Library => true,
+                ProjectOutputType.WinExe => true,
+                _ => false
+            };
+
+        private ProjectOutputType GetProjectOutputType(IProjectRootElement root)
+        {
+            var outputTypeNode = root.GetOutputTypeNode();
```

Comment:
why no switch expression?

Reply:
This is a `ProjectPropertyElement?` so it's not switch-able